src/Microsoft.AspNetCore.Razor/CodeGenerators/CSharpCodeGenerator.cs

```diff
@@ -82,12 +83,22 @@ public override CodeGeneratorResult Generate()
                             csharpCodeVisitor.Accept(Tree.Children);
                         }
                     }
+
+                    BuildAfterExecuteContent(writer, Tree.Children);
                 }
             }
 
             return new CodeGeneratorResult(writer.GenerateCode(), writer.LineMappingManager.Mappings);
         }
 
+        protected virtual void BuildAfterExecuteContent(CSharpCodeWriter writer, IList<Chunk> chunks)
+        {
```

Comment:
Do you mean that we would `DecorateChunks` at `BuildAfterExecuteContent`?

For view component tag helpers, we need to mutate their chunks such that the type name includes the name of the page (ex. `AspNetCore._Views_Index_Cshtml`) since we'll be appending the after execute content within that class. 

This needs to be done before